clusterloader2/util/unmarshaling.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// UnmarshalFileInto unmarshals json or yaml file specified by the given path.
+func UnmarshalFileInto(path string, out interface{}) error {
+	ext := strings.ToLower(filepath.Ext(path))
+	raw, err := ioutil.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("reading file error: %v", err)
+	}
+	if ext == ".yaml" {
```

Comment:
I don't like relying on file extensions.
We already have recognizing decoder in kubernetes repo and we should use that.

pkg/api/legacyscheme.Codecs.universalDecoder is what you're looking for.

Also, I would like to see a test for it.

Reply:
Shouldn't pkg/api/legacyscheme.Codecs.UniversalDecoder be only used for kubernetes configs?
Is it possible to decode any json file with this?

Reply:
What you're going to read with this function?

Reply:
I wanted function that could unmarshal api.Config and object template if possible.
I see that pkg/api/legacyscheme.Codecs.UniversalDecoder is perfect for object template, but I need something more generic for config structure.

Reply:
Then let's not reinvent the wheel and let's use viper the same way the original cluster loader was doing that.
clusterloader/framework/context.go